linera-witty/src/type_traits/implementations/frunk.rs

```diff
@@ -155,3 +165,97 @@ where
         Ok(head_layout + tail_layout)
     }
 }
+
+/// Helper trait used to calculate the size of a heterogeneous list considering internal alignment.
+///
+/// Assumes the maximum alignment necessary for any type is 8 bytes, which is the alignment for the
+/// largest flat types (`i64` and `f64`).
+trait SizeCalculation {
+    /// The size of the list considering the current size calculation starts at the 8-byte
+    /// boundary.
+    const SIZE_STARTING_AT_8_BYTE_BOUNDARY: u32;
```

Comment:
Would this work? `const SIZE_STARTING_AT_BYTE_BOUNDARIES: [u32; 8];`

Reply:
Good idea! That also removes the need for the `calculate_size!` macro. Unfortunately `for` isn't supported in `const` contexts ([yet](https://github.com/rust-lang/rust/issues/87575)?), so I had to add an `unroll_for!` macro instead.